### PR TITLE
Add zNext improvements to z/codegen

### PR DESCRIPTION
**Pull request by dchopra001**
This PR adds the necessary improvements to exploit zNext hardware in the JIT compiler. The commits have been authored by myself, @VermaSh and @r30shah.

### CODE REVIEW

**Comment by dchopra001** [comment]
Tagging @joransiu @r30shah @VermaSh for review. If you could please make sure I've brought over all your changes correctly that would be great! Thanks!

File: port/unix/omrsysinfo.c [hunk 1]
```diff
@@ -1670,6 +1687,8 @@ omrsysinfo_get_s390_processor_feature_name(uint32_t feature)
 		return "vec_e2";
 	case OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL_ENHANCEMENT_FACILITY:
 		return "vec_pde";
+        case OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL_ENHANCEMENT_FACILITY_2:
```
Comment (joransiu) [inline port/unix/omrsysinfo.c:1690]: This case statement doesn't appear to be indented properly.

Comment (dchopra001) [reply port/unix/omrsysinfo.c]: Fixed in 96ffcc84dced3cdde345a701b87d88e62b072433

**Review by joransiu — COMMENTED** [review]
Reviewed the changes, and they generally look good to me.  Thanks @dchopra001!

**Review by r30shah — COMMENTED** [review]
Looks good to me, thanks @dchopra001

File: compiler/control/OMROptions.cpp [hunk 2]
```diff
@@ -2590,6 +2591,7 @@ OMR::Options::jitPreProcess()
    self()->setOption(TR_EnableCodeCacheConsolidation);
 #endif
 
+   self()->setOption(TR_DisableZNext);
```
Comment (fjeremic) [inline compiler/control/OMROptions.cpp:2594]: Are we planning to reenable this in the future?

**Comment by fjeremic** [comment]
Jenkins build all

**Review by VermaSh — CHANGES_REQUESTED** [review]

File: compiler/control/OMROptions.cpp (hunk 2 above)
Comment (joransiu) [reply compiler/control/OMROptions.cpp:2594]: The intention is to disable by default currently -- enabled only with -Xjit:enableZNext.  We will flip the switch in the future.

File: compiler/z/codegen/S390Instruction.cpp [hunk 3]
```diff
@@ -4069,9 +4070,48 @@ uint8_t *
...
+   // Cursor move
+   // update binary length
+   // update binary length estimate error
+   cursor += getOpCode().getInstructionLength();
+   setBinaryLength(cursor - instructionStart);
+   setBinaryEncoding(instructionStart);
+   cg()->addAccumulatedInstructionLengthError(getEstimatedBinaryLength() - getBinaryLength());
+   return cursor;
+   }
+
```
[the remark this answers is not in the document]
Comment (dchopra001) [reply compiler/z/codegen/S390Instruction.cpp]: Seems like this didn't carry over during one of the internal rebases. I've reverted back to the old change here: 63699df1d7d40f0f071a1f50f154bce96ea5f227.

**Review by r30shah — APPROVED** [review]
LGTM

**Review by VermaSh — APPROVED** [review]

**Comment by fjeremic** [comment]
Jenkins build all

**Comment by fjeremic** [comment]
@dchopra001 does this have a downstream dependency or can it be merged independently?

**Review by joransiu — APPROVED** [review]
LGTM

**Comment by dchopra001** [comment]
> @dchopra001 does this have a downstream dependency or can it be merged independently?

We should be okay to merge this independently. Though this will break OpenJ9 until eclipse-openj9/openj9#14179 is merged. Not sure if you meant to ask that instead.

**Comment by joransiu** [comment]
@dchopra001 : The copyright dates will need to be updated to 2022 for this as well.  

@0xdaryl / @fjeremic : This PR and eclipse-openj9/openj9#14179 are both good to go otherwise.  Do you mind reviewing again, and merge if okay?

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
@fjeremic has resolved his only comment so I am concluding from that he is satisfied with the result (and has not raised any other concerns since then).  As other reviews and testing have completed, I will merge this now.

**Pull request merged**